### ISSUE TITLE
TDS 5.3 featureCollection question

### ISSUE THREAD

**Issue by leonid-butenko** [Unidata/tds#193]
Since I moved to the recent version of TDS 5.3-SNAPSHOT (the docker image) my old setup doesn't work anylonger. I used a custom docker build of version 5.0.0b6 before. Now the problem is as following:

here is a description of my collection:

```
  <featureCollection name="Operational Forecast on Model Levels"
    path="operational/forecast/modellevel"
    featureType="GRIB2">

    <metadata inherited="true">
      <serviceName>allServices</serviceName>
      <documentation type="summary">ECMWF operational forecasts on model levels.</documentation>
      <dataFormat>GRIB-2</dataFormat>
    </metadata>

    <collection name="op_fc_ml"
      spec="/data/thredds-testmerged/ecmwf/operational/fc/ml/**/ecmwf_fc_ml_0\.5x0\.5.*\.grb$"
      dateFormatMark="#_0.5x0.5_#yyyyMMdd-HHmm"
      timePartition="file"
      olderThan="15 min"/> 

    <update startup="never" trigger="allow"/> 
    <tdm rewrite="test" rescan="0 0/5 * * * ? *"/>

    <gribConfig datasetTypes="Files" />

  </featureCollection> 
```
For a test purpose i have prepared a list of 4 GRIB-2 files:

```
ecmwf_fc_ml_0.5x0.5_20220101-0000.grb
ecmwf_fc_ml_0.5x0.5_20220102-0000.grb
ecmwf_fc_ml_0.5x0.5_20220103-0000.grb
ecmwf_fc_ml_0.5x0.5_20220104-0000.grb
```

Each GRIB2 file has a dataTime field with a fixed value from the list: [ 6, 12, 18, 24 ]. As long as I put timePartition parameter to"file", I expect the resulting per-file OpenDAP dataset does not depend on other files and to have exactly 4 above values (which is true only if I add only 1 GRIB file to the collection). What I get in my setup with 4 files is an array of concatenated and growing numbers. So that every new file adds exactly 4 new values to all other's datasets. Here is an example:

```
Dataset {
    Float64 time[time = 16];
} operational/forecast/modellevel/op_fc_ml-202201/ecmwf_fc_ml_0.5x0.5_20220104-0000.grb;
---------------------------------------------
time[16]
6.0, 12.0, 18.0, 24.0, 30.0, 36.0, 42.0, 48.0, 54.0, 60.0, 66.0, 72.0, 78.0, 84.0, 90.0, 96.0
```
The array is the same for each of those 4 files.

Additionally if i access my dataset via ncdump, I see that for all 4 files I get the same starting time equal to the first file in the collection. Please have a look at _time:units_ value

```
$>ncdump -v time https://mythredds/thredds/dodsC/operational/forecast/modellevel/op_fc_ml-202201/ecmwf_fc_ml_0.5x0.5_20220104-0000.grb 

netcdf ecmwf_fc_ml_0.5x0.5_20220104-0000 {
dimensions:
        hybrid = 137 ;
        hybrid1 = 1 ;
        lat = 361 ;
        lon = 720 ;
        time = 16 ;
variables:
        float lat(lat) ;
                lat:units = "degrees_north" ;
                lat:_CoordinateAxisType = "Lat" ;
        float lon(lon) ;
                lon:units = "degrees_east" ;
                lon:_CoordinateAxisType = "Lon" ;
        double time(time) ;
                time:units = "Hour since 2022-01-01T00:00:00Z" ;
                time:standard_name = "time" ;
                time:long_name = "GRIB forecast or observation time" ;
                time:calendar = "proleptic_gregorian" ;
                time:_CoordinateAxisType = "Time" ;
```

The docu from here: https://www.unidata.ucar.edu/software/tds/current/reference/collections/Partitions.html says:

```
file: each file is a partition.
...
File Partition
In order to use a file partition, all of the records for a reference time must be contained in a single file. 
The common case is that each file contains all of the records for a single reference time.
```

Do I do something wierd or it is a feature of TDS5 that the files are not handled independently from each other? Pls advise..

**Comment by lesserwhirls** [comment]
It is quite likely you will need to delete any `.ncx4` files associated with your collection (`.gbx9` files should be fine) and let the TDM regenerate them.

**Comment by leonid-butenko** [comment]
> It is quite likely you will need to delete any `.ncx4` files associated with your collection (`.gbx9` files should be fine) and let the TDM regenerate them.

Nope. For this particular test I described above - I started from scratch, i.e. from grib files. No indices, empty cache, etc. And the result - the reference time is always the time of the first file in the collection. I can provide my test data if you would like to run your tests.

**Comment by lesserwhirls** [comment]
I'd be happy to take a look, but it probably won't be before next weekend (as long as I have some free time). If you can zip up your sample files and catalog, that'd be helpful. By the way, the current TDS documentation lives at https://docs.unidata.ucar.edu/tds/5.4/userguide/index.html (the page you specifically referenced is now located at https://docs.unidata.ucar.edu/tds/5.4/userguide/partitions_ref.html).

**Comment by leonid-butenko** [comment]
Thanks a lot for the links to the updated documentation!
I would very much appreciate if you could find time for that exercise. Thanks in advance!

here are the four files from my test dataset: https://drive.google.com/drive/folders/1dlw7nud1sJsVVxR9eniASwu8WfLwZsBl

just copy and paste my featureCollection definition and use any of TDS5 docker images.

**Comment by leonid-butenko** [comment]
Any update on this? Thanks :)

**Comment by lesserwhirls** [comment]
I was able to take a look, and I think this is a change in behavior in netCDF-Java in how collections of non-overlapping forecast (or valid) time values are handled. The reference time is still there (contained in the `reftime` variable, with dimensionality of `time`, and is listed as an auxiliary coordinate for the data variables), but it's not included in the variable's dimensions since there are no overlapping forecast (or valid) times.

The question for Unidata would be if this is change in behavior is a feature or a bug. Since this is a reanalysis with a continuous, non-overlapping set of valid times, I tend to think the new behavior is the better way to represent what is happening in the collection, but I'll leave that call up to others.

**Comment by leonid-butenko** [comment]
Thanks for your feedback! Do you know a way how to draw Unidata's attention to this discussion?

**Comment by haileyajohnson** [comment]
@leonid-butenko We are following, we're just juggling other issues as well! We'll post updates here when we get the chance to talk about it
